### ISSUE TITLE
[Bug] Brickblock powerups still appear when their chance is set to 0%

### ISSUE THREAD

**Issue by OopityDoop** [Coolcord/Level-Headed#27]
![image](https://user-images.githubusercontent.com/65684468/128587982-de826855-c243-4885-aef0-598ddf0cea39.png)
![image](https://user-images.githubusercontent.com/65684468/128587988-49bde094-b6b9-4423-822e-f8b24746191a.png)
https://user-images.githubusercontent.com/65684468/128588008-bcf4acd2-0e05-41a1-885e-7285308ad7d5.mp4

**Comment by Coolcord** [comment]
Thanks for the bug report! But believe it or not, this is actually intended behavior. There are two types of powerups that Level-Headed can allocate: Brick Block Powerups and Question Block Powerups.

To quote the v0.3.6 release notes:

> As of v0.3.6, Level-Headed now generates levels with three passes: objects, powerups, and enemies. This new powerup pass scans the level for hittable blocks that are safe for powerups and distributes items randomly amongst them. This means that powerups and items should now be found more consistently in games.
> 
> It's worth noting that this update doesn't completely solve the powerup issue. As of now, if the level does not generate any hittable blocks in the object pass, the powerup distributor will have no place to put its powerups. This means that there can still be levels without powerups. This issue will be addressed at some point in the future.

To add to this, Level-Headed treats Question Block Powerups differently. **If a level has no Question Blocks, the powerups will be reallocated as Brick Block Powerups so that they can be placed somewhere in the level.** Having some hidden powerups is better than no powerups at all! There are plans to add a proper block pass that places hittable blocks after a level has been generated, but this is still a long ways off.

In short, in this scenario, there were likely no hittable Question Blocks in the level. However, Question Block Powerups were allocated, so they had to be reallocated as Brick Block Powerups. Hopefully this explanation clears up any confusion.

Thanks for the interest in Level-Headed!
